File: 04-algoritmos-basicos/00-fibonacci.js
/*
Ejercicio de entrevista
técnica para el puesto
de programador javascript
La sucesion de fibonacci

salida: 0; 1; 1; 2; 3, 5; 8;
*/

function fibonacci(n){
    let secuencia = [0,1];

    for (let i = 2; i < n; i++) {
        let siguiente_valor = secuencia[i-1] + secuencia[i-2];
        secuencia.push(siguiente_valor);
    }
    return secuencia;
}

let resultado = fibonacci(10);
console.log(resultado);

File: 04-algoritmos-basicos/00-fibonacci.py
"""
Ejercicio de entrevista 
técnica para el puesto 
de programador python jr
La sucesion de fibonacci 

salida: 0; 1; 1; 2; 3, 5; 8;
"""


def fibonacci(n):
    secuencia = [0,1]
    while len(secuencia)<n:
        siguiente_valor = secuencia[-1] + secuencia[-2]
        secuencia.append(siguiente_valor)
    return secuencia

print(fibonacci(10))

def fibo_recursiva(n):
    if n <= 1:
        return n
    else:
        return fibo_recursiva(n-1)+fibo_recursiva(n-2)
    
for i in range(10):
    print(fibo_recursiva(i))

